pkg/xds/server/dataplane_metadata_tracker.go

```diff
@@ -71,3 +96,25 @@ func (d *DataplaneMetadataTracker) OnFetchRequest(context.Context, *envoy.Discov
 
 func (d *DataplaneMetadataTracker) OnFetchResponse(*envoy.DiscoveryRequest, *envoy.DiscoveryResponse) {
 }
+
+func (d *DataplaneMetadataTracker) registerDataplane(dp *core_mesh.DataplaneResource) error {
+	if d.environmentType != core.UniversalEnvironment {
+		return nil
+	}
```

Comment:
I think this if should be gone and we should register dataplane only if send it over metadata